legend-engine-pure-code-compiled-core/src/main/resources/core/pure/binding/executionPlan/executionPlan_generation.pure

```diff
@@ -90,24 +93,48 @@ function <<access.private>> meta::external::shared::format::executionPlan::getFu
 
 function <<access.private>> meta::external::shared::format::executionPlan::externalizeFunctionProcessor(fe:FunctionExpression[1], state:ExternalFormatPlanGenerationState[1], extensions : Extension[*], debug:DebugContext[1]):ExecutionNode[1]
 {
-   let children        = $fe.parametersValues->evaluateAndDeactivate()->at(0)->processValueSpecification($state, $extensions, $debug)->toOneMany();
-
+   let parameters      = $fe.parametersValues->evaluateAndDeactivate();
+   let children        = $parameters->at(0)->processValueSpecification($state, $extensions, $debug)->toOneMany();
+   
    let inScopeVars     = $state.inScopeVars;
+   
+   let tree            = if($parameters->size() == 3,  // TODO: This condition should not be needed once tree is mandated
+                            | $parameters->at(2)->byPassValueSpecificationWrapper()->match([
+                                i:InstanceValue[1]            | $i.values->cast(@RootGraphFetchTree<Any>)->first(),
+                                s:SimpleFunctionExpression[1] | assert($s.func == checked_RootGraphFetchTree_1__Binding_1__RootGraphFetchTree_1_, | 'Please provide a tree or use a checked function as a parameter for externalize');
+                                                                let valueTree  = $s.parametersValues->at(0)->match([
+                                                                                                                v:VariableExpression[1] | $inScopeVars->get($v.name).values,
+                                                                                                                i:InstanceValue[1]      | $i.values
+                                                                                                             ])->cast(@RootGraphFetchTree<Any>)->toOne();
+                                                                let binding    = $s.parametersValues->at(1)->match([
+                                                                                                                v:VariableExpression[1] | $inScopeVars->get($v.name).values,
+                                                                                                                i:InstanceValue[1]      | $i.values
+                                                                                                             ])->cast(@Binding)->toOne();
```

Comment:
Doesn't reactivate function do this resolving for us?

Reply:
We will still need a match clause, also had tried reactivate for variable though casting caused some issues and I believe we anyway try to avoid reactivate so went with this